lib/customer-deposit-wallet-pure/haskell/Cardano/Wallet/Deposit/Pure/RollbackWindow.hs

```diff
@@ -0,0 +1,70 @@
+{-# LANGUAGE ScopedTypeVariables #-}
+
+module Cardano.Wallet.Deposit.Pure.RollbackWindow where
+
+if' :: Bool -> a -> a -> a
+if' True thn els = thn
+if' False thn els = els
+
+data RollbackWindow time = RollbackWindowC
+    { finality :: time
+    , tip :: time
+    }
+
+member :: Ord time => time -> RollbackWindow time -> Bool
+member time w = finality w <= time && time <= tip w
+
+singleton :: Ord time => time -> RollbackWindow time
+singleton time = RollbackWindowC time time
+
+rollForward
+    :: Ord time
+    => time
+    -> RollbackWindow time
+    -> Maybe (RollbackWindow time)
+rollForward newTip w =
+    if tip w <= newTip
```

Comment:
shouldn't this be a Nothing in case of equality ?

Reply:
It's a choice, but I am also in favor of `<`. Will change.

`<` will play nicer with `member` via the following property:

```agda
∀ {time} {{_ : Ord time}} {{@0 _ : IsLawfulOrd time}}
  (t : time)
  (w : RollbackWindow)
→ isJust (rollForward t w) ≡ True
→ member t w ≡ False
```